pkg/drivers/cdp/dom/helpers.go

```diff
@@ -0,0 +1,285 @@
+package dom
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"golang.org/x/net/html"
+	"strings"
+	"time"
+
+	"github.com/MontFerret/ferret/pkg/drivers/cdp/eval"
+	"github.com/MontFerret/ferret/pkg/drivers/cdp/templates"
+	"github.com/MontFerret/ferret/pkg/drivers/common"
+	"github.com/MontFerret/ferret/pkg/runtime/values"
+
+	"github.com/PuerkitoBio/goquery"
+	"github.com/mafredri/cdp"
+	"github.com/mafredri/cdp/protocol/dom"
+	"github.com/mafredri/cdp/protocol/page"
+	"github.com/mafredri/cdp/protocol/runtime"
+)
+
+var emptyExpires = time.Time{}
+
+func parseAttrs(attrs []string) *values.Object {
```

Comment:
I don’t quite understand what is going on here

Reply:
CDP returns HTML attributes as an array of strings of key-value pairs. This methods parses the array and converts it into an object.